src/metabase/query_processor/middleware/resolve_referenced.clj

```diff
@@ -0,0 +1,75 @@
+(ns metabase.query-processor.middleware.resolve-referenced
+  (:require [metabase.models.card :refer [Card]]
+            [metabase.query-processor.middleware
+             [resolve-fields :as qp.resolve-fields]
+             [resolve-source-table :as qp.resolve-tables]]
+            [metabase.util.i18n :refer [deferred-tru]]
+            [schema.core :as s]
+            [toucan.db :as db]
+            [weavejester.dependency :as dep])
+  (:import clojure.lang.ExceptionInfo))
+
+(defn tags-referenced-cards
+  "Returns Card instances referenced by the given native `query`."
+  [query]
+  (->> (get-in query [:native :template-tags])
```

Comment:
why doesn't this use `query->template-tags` & `query->tag-card-ids`?

Reply:
Only because of the order I wrote these functions in. :wink:

Fixed.